.travis/upload_to_s3.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -ex
```

Comment:
shouldn't you also `set +ex` after the script ends?

Reply:
No, that is only necessary when a script is sourced (as `set_tag.sh` is) because then it's running in the context of the global travis shell. This is normally executed in a sub-shell.